shared/README.md

```diff
@@ -0,0 +1,77 @@
+# Shared
+
+The `libretime_shared` package contains reusable functions and classes for the Libretime project.
+
+## Usage
+
+This library assumes that:
+
+- You will use [`Click`](https://github.com/pallets/click) to build a CLI for your app.
+- You will use [`Loguru`](https://github.com/delgan/loguru) to log messages from your app.
+- You will use [`Pydantic`](https://github.com/samuelcolvin/pydantic/) to validate objects in your app.
+
+### Configuration
+
+First define a schema for your configuration in order to validate it. A schema is a class that inherit from `pydantic.BaseModel`. Some existing schemas can be reused such as `libretime_shared.config.RabbitMQ` or `libretime_shared.config.Database`.
+
+Load your configuration using a subclass of `libretime_shared.config.BaseConfig`.
+
+```py
+from pydantic import BaseModel
+
+from libretime_shared.config import RabbitMQ, BaseConfig
+
+class Analyzer(BaseModel):
+    bpm_enabled: bool = False
+    bpm_track_max_length: int
+
+class Config(BaseConfig):
+    rabbitmq: RabbitMQ
+    analyzer: Analyzer
+
+config = Config(filepath="/opt/libretime/analyzer/config.yml")
```

Comment:
How does this handle common configuration such as database? Does it need to be duplicated into all applications that access it?

Reply:
Similar to the RabbitMQ field, you declare the config requirement in your app, and load it:
```py
from libretime_shared.config import BaseConfig, BaseModel, Database

class Analyzer(BaseModel):
    bpm_enabled: bool = False
    bpm_track_max_length: int

class Config(BaseConfig):
    database: Database
    analyzer: Analyzer

config = Config(filepath="/opt/libretime/analyzer/config.yml")
```

I assumed that only the MQ and Database Models are shared across the different apps, otherwise each app will have to define there own Configuration fields.

Reply:
But that example uses `/opt/libretime/analyzer/config.yml`, so my reading would be that each app has their own configuration file. Is that correct? Otherwise do they all use the same file and ignore the sections that are not defined for their particular schema?

Reply:
Ah ok, yes, the example file path is not realistic. All the configuration will be stored in a single file and each app will have to load from the same file.

This makes me think that a helper function / default field to get the config from the default path can be a good idea, so we have a single place where the default configuration path is.

Reply:
I haven't tested it though !

Reply:
Maybe also use `/etc/libretime/config.yml` in the example documentation so that it is clearer that it should be the same file throughout

Reply:
Done

Reply:
A helper function would be very useful, as having the option to specify the config file as an environment variable is useful for the Docker side of things

Reply:
I'm happy to merge as is and then helper functions/extensions can be handled as improvements

Reply:
Ok, I'll do this in another PR